### DISCUSSION TITLE
new linter with ruleguard load error

### DISCUSSION THREAD

**Discussion by oscarliu2019** [golangci/golangci-lint#3467] [Q&A]
I want to add a new linter to the golangci-lint. Following the guide, i add the a linter called checkname
```go
func NewCheckName() *goanalysis.Linter {
	return goanalysis.NewLinter(
		"checkname",
		"check the file name and the package name",
		[]*analysis.Analyzer{analyzer.Analyzer},
		nil,
	).WithLoadMode(goanalysis.LoadModeNone)
}
```

in the package/golinters/checkname.go, and then i add it to the manager.go

```go
		linter.NewConfig(golinters.NewBodyclose()).
			WithSince("v1.18.0").
			WithLoadForGoAnalysis().
			WithPresets(linter.PresetPerformance, linter.PresetBugs).
			WithURL("https://github.com/timakin/bodyclose"),
		linter.NewConfig(golinters.NewCheckName()).WithPresets(linter.PresetStyle).WithURL(""),
		linter.NewConfig(golinters.NewContainedCtx()).
			WithSince("1.44.0").
			WithPresets(linter.PresetStyle).
			WithURL("https://github.com/sivchari/containedctx"),
```
and
```go
	enabledByDefault := map[string]bool{
		golinters.NewGovet(nil).Name():                  true,
		golinters.NewErrcheck(errcheckCfg).Name():       true,
		golinters.NewStaticcheck(staticcheckCfg).Name(): true,
		golinters.NewUnused(unusedCfg).Name():           true,
		golinters.NewGosimple(gosimpleCfg).Name():       true,
		golinters.NewIneffassign().Name():               true,
		golinters.NewTypecheck().Name():                 true,
		golinters.NewCheckName().Name():                 true,
	}
```
i also add the new linter to the .golangci.yml
```xml
linters:
  disable-all: true
  enable:
    - checkname
```

however, when i build the linters and run the command
`go build -o linter ./cmd/golangci-lint/main.go`
and test the file
`./linter run ./test`, the linter have error message
**WARN [runner] Can't run linter checkname: ruleguard: load rules: both -e and -rules flags are empty 
ERRO Running error: 1 error occurred:
        * can't run linter checkname: ruleguard: load rules: both -e and -rules flags are empty**

I wonder how to fix it, thankyou

**Comment by Antonboom** [comment]
Hi!

First of all, only small set of linters are allowed to be enabled by default. Please remove
```go
golinters.NewCheckName().Name():                 true,
```

The second one, can you attach link to the linter source code or to MR in golangci-lint with your integration?
Judging by the name, there is a feeling that a similar linter already exists.

Thanks
